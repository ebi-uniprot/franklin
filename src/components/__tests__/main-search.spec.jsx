import { render, fireEvent } from '@testing-library/react';
import MainSearch from '../main-search';

describe('MainSearch component', () => {
  const handleSubmit = jest.fn();
  const handleChange = jest.fn();
  const props = {
    onChange: handleChange,
    onSubmit: handleSubmit,
  };
  const namespaces = { one: 'One', two: 'Two', three: 'Three' };

  test('should render', () => {
    const { asFragment } = render(<MainSearch {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render with namespaces selector', () => {
    const { asFragment } = render(
      <MainSearch {...props} namespaces={namespaces} selectedNamespace="one" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('should submit the search term', () => {
    const { getByTestId } = render(<MainSearch {...props} />);
    fireEvent.submit(getByTestId('main-search-form'));
    expect(handleSubmit).toHaveBeenCalled();
  });

  test('should detect the search term', () => {
    const { getByTestId } = render(<MainSearch {...props} />);
    fireEvent.change(getByTestId('main-search-input'), {
      target: { value: 'foo' },
    });
    expect(handleChange).toHaveBeenCalledWith('foo');
  });

  test('should set searchTerm', () => {
    const { getByTestId } = render(<MainSearch {...props} searchTerm="blah" />);
    expect(getByTestId('main-search-input').value).toBe('blah');
  });
});
