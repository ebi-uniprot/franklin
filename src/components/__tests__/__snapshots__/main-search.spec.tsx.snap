// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSearch component should render 1`] = `
<DocumentFragment>
  <form
    aria-label="Main search"
    class="main-search"
    style="--main-button-color: #00639a;"
  >
    <div
      class="main-search__input-container"
    >
      <input
        aria-label="Text query"
        type="text"
        value=""
      />
    </div>
    <button
      class="button primary"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;

exports[`MainSearch component should render with namespaces selector 1`] = `
<DocumentFragment>
  <form
    aria-label="Main search"
    class="main-search"
    style="--main-button-color: #00639a;"
  >
    <div
      class="dropdown-container"
    >
      <button
        class="button primary dropdown"
        type="button"
      >
        One
      </button>
      <div
        class="dropdown-menu"
      />
    </div>
    <div
      class="main-search__input-container"
    >
      <input
        aria-label="Text query in one"
        type="text"
        value=""
      />
    </div>
    <button
      class="button primary"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
